### ISSUE TITLE
BigQuery: Joins in the query builder generate invalid table aliases

### ISSUE THREAD

**Issue by nevesb** [metabase/metabase#14933]
**Describe the bug**
Hi.
I'm brazilian and when we create tables or fields, we gave portuguese names to be more familiar to ours contrubutors.
Recently we upgraded Metabase from 0.36 to 0.38, and a lot of questions stoped to work.

When I put the generated query directly on BigQuery I got an error related to field names generated by Metabase.

The error is related with the usage of accents, space, etc...

Error: Invalid field name "Organização__customer_success". Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 128 characters long.

Generated Query:

```sql
SELECT 
  `Organização`.`customer_success` AS `Organização__customer_success`, 
  count(distinct `Organização`.`name`) AS `count`
FROM 
  `data_mart.zendesk_tickets`
LEFT JOIN 
  `data_mart.users` `Email do solicitante` ON `data_mart.zendesk_tickets`.`requester_email` = `Email do solicitante`.`email` 
  LEFT JOIN `data_mart.salesforce_usergroups` `Organização` ON `Email do solicitante`.`user_group` = `Organização`.`company`
WHERE (`data_mart.zendesk_tickets`.`motivo` = 'falha'
   AND `Organização`.`name` IS NOT NULL AND (`Organização`.`name` <> ''
    OR `Organização`.`name` IS NULL) AND (`Organização`.`mrr` <> 0 OR `Organização`.`mrr` IS NULL))
GROUP BY 
  `Organização__customer_success`
ORDER BY 
  `count` DESC, 
  `Organização__customer_success` ASC
  ```

**To Reproduce**
1. At version 0.36 rename a table at Data Model screen with a name with acents, like: `Organização`
2. Upgrade Metabase to 0.38
3. Create a question joining the renamed table with any other.
4. Click to visualize
5. The query will load forever
6. Click to see the native query
7. Copy the generated native query and paste at BigQuery
8. See the error like: Invalid field name "Organização__customer_success". Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 128 characters long.

**Expected behavior**
When I click to visualize at step 4, the chart are showed.

**Information about your Metabase Installation:**

```json
{
  "browser-info": {
    "language": "pt-BR",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "4.19.150+",
    "user.language": "en",
    "user.timezone": "UTC"
  },
  "metabase-info": {
    "databases": [
      "bigquery",
      "googleanalytics",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.25-google-log"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.6.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-02-16",
      "tag": "v0.38.0",
      "branch": "release-x.38.x",
      "hash": "471aa51"
    },
    "settings": {
      "report-timezone": "America/Sao_Paulo"
    }
  }
}
```

**Severity**
Its blocking the upgrade to 0.38. A lot of questions in my Metabase stoped to work when I upgraded to 0.38.
I did the rollback to 0.36 but I realy cant enjoy the updates until it works.

**Comment by flamber** [comment]
Step 5 - loading spinner "forever" is the same problem as #14918

**Comment by nevesb** [comment]
Only the fact of load spin forever fact.
The cause of the error might me diferent I guess

**Comment by flamber** [comment]
@nevesb Can you try upgrading to 0.37.9? It's a very big jump from 0.36.x to 0.38.0, and I'm trying to narrow down when the problem started.

**Comment by nevesb** [comment]
> @nevesb Can you try upgrading to 0.37.9? It's a very big jump from 0.36.x to 0.38.0, and I'm trying to narrow down when the problem started.

I did it. And waiting for a patch to resolve the 0.38

**Comment by flamber** [comment]
@nevesb So you're saying that it works on 0.37.9?

**Comment by nevesb** [comment]
> @nevesb So you're saying that it works on 0.37.9?

Yes, at 0.37.9 its working fine!

**Comment by nevesb** [comment]
The same query above generated at 0.37.9:
```sql
SELECT
  `Organização`.`customer_success` AS `customer_success`,
  COUNT(DISTINCT `Organização`.`name`) AS `count`
FROM
  `data_mart.zendesk_tickets`
  LEFT JOIN `data_mart.users` `Email do solicitante` ON `data_mart.zendesk_tickets`.`requester_email` = `Email do solicitante`.`email`
  LEFT JOIN `data_mart.salesforce_usergroups` `Organização` ON `Email do solicitante`.`user_group` = `Organização`.`company`
WHERE
  (`data_mart.zendesk_tickets`.`motivo` = 'falha' 
  AND `Organização`.`name` IS NOT NULL AND (`Organização`.`name` <> ''
      OR `Organização`.`name` IS NULL) AND (`Organização`.`mrr` <> 0 OR `Organização`.`mrr` IS NULL))
GROUP BY
  `customer_success`
ORDER BY
  `count` DESC,
  `customer_success` ASC
  ```

**Comment by nevesb** [comment]
The diference here is the generated colunm name:
At 0.38:
`Organização__customer_success`
At 0.37.9:
`customer_success`

You can use acents renaming table names, but not to colunm names.

**Comment by shihyuuuuuuu** [comment]
I also encountered this bug, and I think the situation is similiar.
**Error:**
`Invalid field name "Top 20 Im Prs__slot_account". Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 128 characters long.`

Metabase generated query like this:
```sql
SELECT `Top 20 Im Prs`.`slot_account` AS `Top 20 Im Prs__slot_account`
FROM
...
...
```
This query caused error in BigQuery because there are spaces in the SELECT field.
I met this issue yesterday and I think it's also caused by the new release of Metabase.
Thanks!

> My Metabase versoin: v0.38.0.1

**Comment by danhawkins** [comment]
I have this issue also `Invalid field name "Client Order Timings - Client Order__finished_at_utc". Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 128 characters long.
`

**Comment by JGrubb** [comment]
For completeness - this only affects queries that explicitly join tables in the UI. If tables are filtered on columns in related tables (via the FK connections in the data model) the generated table name is valid -

```
...
LEFT JOIN `staging.accounts_users` `accounts_users__via__owner_id` ON `staging.accounts_projects`.`owner_id` = `accounts_users__via__owner_id`.`user_id`
...
```

**Comment by JGrubb** [comment]
Last thing - we have a MySQL DB connected to the same instance and it also generates table aliases taken from the name given in the UI, eg 

```
LEFT JOIN `staging.accounts_users` `Accounts: Users - Owner`
```

so this does not appear to be specific to the BQ driver.  I guess only BQ chokes on these types of table aliases.

**Comment by camsaul** [comment]
Fixed by #15023
